deptofdefense/move.mil
# Downgrade Ruby and Modify Configuration Files

## Checklist

I have…


## Summary of Changes

This pull request downgrades the project's Ruby version to 2.3.4 which is the most recent version available on AWS.

Additionally, AWS deployments prefer configuration files like `config/database.yml` and `config/secrets.yml` be committed to the repo. That preference precipitated the addition of the [dotenv-rails](https://github.com/bkeepers/dotenv) to the project so that we might commit the necessary configuration files to the repo without including confidential information.

Henceforth, in development, environment variables will reside in a `.env` file:

```
export MY_SECRET_API_KEY=f867dde900d6b42bf6babc81270da35fb0418c
```

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout downgrade-ruby`,
1. install Ruby 2.3.4 (e.g. `rbenv install 2.3.4`),
1. install Bundler (`gem install bundler`),
1. `bundle install --full-index`,
1. `bin/rails setup`,
1. run `bin/rails server` and verify the site loads,
1. run `bin/rake` to run the test suite, and
1. run `bin/rubocop` and `bin/brakeman` to run static code analysis.
